src/goto-cc/compile.cpp

```diff
@@ -135,56 +136,113 @@ bool compilet::doit()
     warnings_before;
 }
 
+enum class file_typet
+{
+  NO_SUCH_FILE,
+  UNKNOWN,
+  SOURCE_FILE,
+  NORMAL_ARCHIVE,
+  THIN_ARCHIVE,
+  GOTO_BINARY,
+  ELF_OBJECT
+};
+
+static file_typet detect_file_type(const std::string &file_name)
+{
+  // first of all, try to open the file
+  std::ifstream in(file_name);
+  if(!in)
+    return file_typet::NO_SUCH_FILE;
```

Comment:
Opening a file can fail for other reasons than the file not existing, although as far as I'm aware there's no portable way to detect this in C++11 so I guess this is fine.

Reply:
I have renamed the enum value and also made the warning message more detailed.